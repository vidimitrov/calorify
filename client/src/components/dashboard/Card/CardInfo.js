import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Wrapper = styled.div`
  color: #f0cd61;
  margin: 0px 20px;
  flex-grow:1;

  @media (max-width: 420px) {
    min-width: 105px;
  }
`;

const Text = styled.h3`
  margin: 0;
  color: #c7b26e;

  @media (max-width: 420px) {
    font-size: 16px;
  }
`;

const Calories = styled.h1`
  margin: 0;

  @media (max-width: 420px) {
    font-size: 18px;
  }
`;

const CardInfo = ({ text, calories }) => (
  <Wrapper>
    <Calories>
      {calories}
      {' '}
      kCal
    </Calories>
    <Text>{text}</Text>
  </Wrapper>
);

CardInfo.propTypes = {
  text: PropTypes.string.isRequired,
  calories: PropTypes.number.isRequired,
};

export default CardInfo;
