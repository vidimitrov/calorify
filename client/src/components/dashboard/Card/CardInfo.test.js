import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import CardInfo from './CardInfo';

it('Dashboard/Card/CardInfo component renders correctly', () => {
  const tree = renderer.create(<CardInfo name="Chicken with rice" calories={320} />).toJSON();
  expect(tree).toMatchSnapshot();
});
