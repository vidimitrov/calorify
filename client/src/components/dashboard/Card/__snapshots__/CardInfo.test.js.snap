// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard/Card/CardInfo component renders correctly 1`] = `
.c0 {
  color: #f0cd61;
  margin: 0px 20px;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c1 {
  margin: 0;
}

.c2 {
  margin: 0;
  color: #c7b26e;
}

@media (max-width:420px) {
  .c0 {
    min-width: 70px;
    margin: 0px 15px;
    width: 200px;
  }
}

@media (max-width:420px) {
  .c1 {
    font-size: 18px;
  }
}

@media (max-width:420px) {
  .c2 {
    font-size: 16px;
    word-break: break-all;
  }
}

<div
  className="c0"
>
  <h1
    className="c1"
    name="card-info-title"
  />
  <h3
    className="c2"
    name="card-info-subtitle"
  />
</div>
`;
